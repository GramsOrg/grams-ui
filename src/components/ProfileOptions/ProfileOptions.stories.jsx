import React from 'react';

import { action } from "@storybook/addon-actions";

import ProfileOptions from './ProfileOptions.tsx';

export default {
  title: 'Onboarding/ProfileOptions',
  component: ProfileOptions,
  parameters: {
    componentSubtitle: 'ProfileOptions is sugar for <SelectionList />',
    storyshots: { disable: false },
  }
};

const Template = ({ ...props }) => {
  return (
    <ProfileOptions
      onSelect={action("Trigger onSelect()")}
      {...props}
    />
  );
};

export const Default = Template.bind({});
Default.args = {
}
