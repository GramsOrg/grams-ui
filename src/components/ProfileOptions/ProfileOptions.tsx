import React from 'react';

import { useTranslation } from 'react-i18next';

import '../../main.css';
import SelectionList from '../SelectionList';
import { SelectionOption } from '../SelectionList/SelectionList';

export interface IProfileOptionsProps {

  /**
   * A callback function called when an option is selected.
   */
  onSelect?: (id: string) => void;
}

const defaultProps = {
};

const ProfileOptions = (props: IProfileOptionsProps) => {

  const { t } = useTranslation();

  const options: SelectionOption[] = [
    {
      id: 'new',
      title: t('ProfileOptions.new.title'),
      description: t('ProfileOptions.new.description'),
      icon: 'user plus',
      color: 'black'
    }, {
      id: 'backup',
      title: t('ProfileOptions.backup.title'),
      description: t('ProfileOptions.backup.description'),
      icon: 'upload',
      color: 'black'
    }, {
      id: 'seed',
      title: t('ProfileOptions.seed.title'),
      description: t('ProfileOptions.seed.description'),
      icon: 'pencil',
      color: 'black'
    }, {
      id: 'ledger',
      title: t('ProfileOptions.ledger.title'),
      description: t('ProfileOptions.ledger.description'),
      icon: 'usb',
      color: 'black'
    }, 
  ]

  return (
    <SelectionList options={options} onSelect={option => props?.onSelect?.(option)} />
  );
};

ProfileOptions.defaultProps = defaultProps;

export default ProfileOptions;
