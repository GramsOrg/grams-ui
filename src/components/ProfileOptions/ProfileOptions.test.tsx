/**
 * @jest-environment jsdom
 */

import React from 'react';
import { describe, expect, test } from '@jest/globals';
import { render } from '@testing-library/react';

import ProfileOptions from './ProfileOptions';

describe('ProfileOptions Component', () => {
  test('it should match the snapshot', () => {
    const { asFragment } = render(<ProfileOptions />);
    expect(asFragment()).toMatchSnapshot();
  });
});
