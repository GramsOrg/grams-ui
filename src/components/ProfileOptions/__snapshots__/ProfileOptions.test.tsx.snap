// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileOptions Component it should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui selection very relaxed list ltr"
    role="list"
  >
    <div
      class="item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="black user plus big icon middle"
      />
      <div
        class="content ltr-content"
      >
        <h2
          class="header"
        >
          Create new profile
        </h2>
        <div
          class="description"
        >
          Start a new profile from scratch
        </div>
      </div>
    </div>
    <div
      class="item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="black upload big icon middle"
      />
      <div
        class="content ltr-content"
      >
        <h2
          class="header"
        >
          Import profile backup
        </h2>
        <div
          class="description"
        >
          Import a .grams backup file
        </div>
      </div>
    </div>
    <div
      class="item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="black pencil big icon middle"
      />
      <div
        class="content ltr-content"
      >
        <h2
          class="header"
        >
          Import profile seed
        </h2>
        <div
          class="description"
        >
          Import your profile using the 24-word seed
        </div>
      </div>
    </div>
    <div
      class="item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="black usb big icon middle"
      />
      <div
        class="content ltr-content"
      >
        <h2
          class="header"
        >
          Use Ledger wallet
        </h2>
        <div
          class="description"
        >
          Pair Grams with your ledger wallet
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
